R/CDISCFilteredData.R

```diff
@@ -160,7 +160,9 @@ CDISCFilteredData <- R6::R6Class( # nolint
       varnames <- self$get_filtered_dataset(dataname)$get_filterable_varnames()
       parent_dataname <- self$get_parentname(dataname)
       parent_varnames <- if (length(parent_dataname) > 0) {
-        super$get_filterable_varnames(parent_dataname)
+        # cannot call get_filterable_varnames on the parent filtered_dataset in case
+        # some of its variables are set to be non-filterable
+        self$get_filtered_dataset(parent_dataname)$get_supported_filter_varnames()
```

Comment:
This is the only place (other than from within the class itself) where we are calling `$get_supported_filter_varnames()`. Can we call S3 method directly and make this method private? Can we remove this method completely and replace all occurrences with S3 method?
Having two separate methods `get_supported_filter_varnames` and `get_filterable_varnames` might be confusing and somewhat complex.